docs/introduction/basic_concepts.md

```diff
@@ -0,0 +1,44 @@
+# Basic Concepts
+
+This section outlines Jellyfish building blocks.
+It is highly recommended to read it before going further as terms
+introduced here are used in subsequent chapters.
+
+## Room
+
+A room is a container for peers and components.
+Its meaning depends on how it is used.
+
+In case of video conferencing system, one Jellyfish room will represent one real
+room or channel where multiple people can talk to each other.
+
+In case of broadcasting system, one room will be a container for two components:
+one responsible for receiving stream from a streaming host (e.g. via RTMP) and
+the other one responsible for converting this stream into HLS playlist ready to be
+broadcasted via CDN.
+
+
+## Component
+
+A component is a server side process that publishes or subscribes to tracks.
```

Comment:
I really don't like `entity` word because it really hard to image what an entity means. The only place where I used it was when explaining `peer` as I couldn't find better word